packages/web-components/fast-foundation/src/number-field/number-field.template.ts

```diff
@@ -0,0 +1,75 @@
+import { html, ref, slotted } from "@microsoft/fast-element";
+import { endTemplate, startTemplate } from "../patterns";
+import { NumberField } from "./number-field";
+
+/**
+ * The template for the {@link @microsoft/fast-foundation#(NumberField:class)} component.
+ * @public
+ */
+export const NumberFieldTemplate = html<NumberField>`
+    <template
+        class="
+            ${x => (x.readOnly ? "readonly" : "")}
+        "
+    >
+        <label
+            part="label"
+            for="control"
+            class="${x =>
+                x.defaultSlottedNodes && x.defaultSlottedNodes.length
+                    ? "label"
+                    : "label label__hidden"}"
+        >
+            <slot ${slotted("defaultSlottedNodes")}></slot>
+        </label>
+        <div class="root" part="root">
+            ${startTemplate}
+            <input
+                class="control"
+                part="control"
+                id="control"
+                @input="${x => x.handleTextInput()}"
+                @change="${x => x.handleChange()}"
+                ?autofocus="${x => x.autofocus}"
+                ?disabled="${x => x.disabled}"
+                list="${x => x.list}"
+                maxlength="${x => x.maxlength}"
+                minlength="${x => x.minlength}"
+                placeholder="${x => x.placeholder}"
+                ?readonly="${x => x.readOnly}"
+                ?required="${x => x.required}"
+                size="${x => x.size}"
+                :value="${x => x.value}"
+                type="text"
```

Comment:
@robarbms Are there any potential implications with having this be articulated as text semantically and via AT but having it behave strictly as a number?

Reply:
I used text type rather than number to avoid limitations of number. You can not use float values unless your steps are set to a float value(not 1.0, it needs to be a fractional like 0.1). I do number validation on the component side rather than using the in built validation of type number.

Reply:
Thanks! Given that shift, I wonder if we should add support for `inputmode` as an attribute here with only numeric types? Research on that here for input with type text for numbers similar to the choice we're making: https://technology.blog.gov.uk/2020/02/24/why-the-gov-uk-design-system-team-changed-the-input-type-for-numbers/

Reply:
I've updated the inputmode. I left off pattern because I already had validation being handled on the valueChanged method using parseFloat. Their pattern seemed incomplete for data statistics and trying to maintain a pattern could be potentially problematic.